lib/logster/base_store.rb

```diff
@@ -65,9 +71,22 @@ def report(severity, progname, msg, opts = {})
 
       return if ignore && ignore.any? { |pattern| message =~ pattern}
 
-      save message
+      similar = nil
+
+      if Logster.config.allow_grouping
+        recent = latest(limit: 20, severity: [severity])
```

Comment:
this ain't gonna cut it, its just gonna be full of edge cases and risks performing real bad if there is a flood 

The store needs a second redis hash that goes from message.grouping_key -> message.key, and you need to ensure you invalidate the hash correctly.